From b4d2c75d394d1b178264a3e1e6721b4f77fa1eac Mon Sep 17 00:00:00 2001
From: Luc Maisonobe <luc@apache.org>
Date: Wed, 8 Apr 2015 17:12:32 +0200
Subject: [PATCH] Fixed wrong handling of a corner case.

The problem occurs in very difficult to reproduce cases, as it is really
a corner case. When the vanishing cut-subhyperplane occurs during a
Difference operation, the fixed node still has a null cut-subhyperplane.
This was not proerply handled.

JIRA: MATH-1162
---
 .../apache/commons/math3/geometry/partitioning/BSPTree.java    | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/main/java/org/apache/commons/math3/geometry/partitioning/BSPTree.java b/src/main/java/org/apache/commons/math3/geometry/partitioning/BSPTree.java
index 26ae45fef5..53952189c2 100644
--- a/src/main/java/org/apache/commons/math3/geometry/partitioning/BSPTree.java
+++ b/src/main/java/org/apache/commons/math3/geometry/partitioning/BSPTree.java
@@ -707,6 +707,9 @@ public void insertInTree(final BSPTree<S> parentTree, final boolean isPlusChild,
                     plus      = fixed.plus;
                     minus     = fixed.minus;
                     attribute = fixed.attribute;
+                    if (cut == null) {
+                        break;
+                    }
                 }
 
             }
